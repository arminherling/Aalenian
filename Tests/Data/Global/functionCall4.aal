def inner(x: ref i32) 
{ 
}

def outer()
{
    x = 20
    inner(ref x)
}
