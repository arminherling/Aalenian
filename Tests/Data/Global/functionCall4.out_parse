FunctionDefinitionStatement: {
    NameExpression: inner
    ParametersNode(1): {
        ParameterNode: {
            NameExpression: x
            Ref: true
            TypeNode: i32
        }
    }
    BlockNode(0): {
    }
}
FunctionDefinitionStatement: {
    NameExpression: outer
    ParametersNode(0): {
    }
    BlockNode(2): {
        AssignmentStatement: {
            Left: {
                NameExpression: x
            }
            Right: {
                NumberLiteral: 20
            }
        }
        ExpressionStatement: {
            FunctionCallExpression: {
                NameExpression: inner
                ArgumentsNode(1): {
                    ArgumentNode: {
                        Ref: true
                        NameExpression: x
                    }
                }
            }
        }
    }
}
