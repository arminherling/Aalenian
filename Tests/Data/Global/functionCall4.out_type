TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: (ref i32)->()
    Name: inner
    Parameters(1): {
        Parameter: {
            Type: ref i32
            Name: x
        }
    }
    ReturnType: {
        Type: void
    }
    Body(0): {
    }
}
TypedFunctionDefinitionStatement: {
    TypeKind: function
    Type: ()->()
    Name: outer
    Parameters(0): {
    }
    ReturnType: {
        Type: void
    }
    Body(2): {
        TypedAssignmentStatement: {
            Type: i32
            Left: {
                TypedVariable: {
                    Type: i32
                    Name: x
                }
            }
            Right: {
                I32Value: {
                    Type: i32
                    Value: 20
                }
            }
        }
        TypedExpressionStatement: {
            Type: void
            Expression: {
                TypedFunctionCallExpression: {
                    Type: void
                    Name: inner
                    Arguments(1): {
                        TypedReferenceOfExpression: {
                            Type: ref i32
                            Expression: {
                                TypedVariable: {
                                    Type: i32
                                    Name: x
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
