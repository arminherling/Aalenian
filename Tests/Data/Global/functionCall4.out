FunctionDefinitionStatement: {
    Name: inner
    Parameters(1): {
        Parameter: {
            Name: x
            Ref: true
            Type: i32
        }
    }
    Block(0): {
    }
}
FunctionDefinitionStatement: {
    Name: outer
    Parameters(0): {
    }
    Block(2): {
        AssignmentStatement: {
            Left: {
                Name: x
            }
            Right: {
                Number: 20
            }
        }
        ExpressionStatement: {
            FunctionCall: {
                Name: inner
                Arguments(1): {
                    Argument: {
                        Ref: true
                        Name: x
                    }
                }
            }
        }
    }
}
